### PR TITLE
Improved `pachctl mount` support on macOS

### PR DESCRIPTION
**Pull request by ysimonson**
A few improvements to `pachctl mount` support on macOS:

* Use `golang.org/x/sys/unix` instead of `syscall` where possible. [It is the successor to syscall](https://docs.google.com/document/d/1QXzI9I1pOfZPujQzxhyRy6EeHYTQitKKjHfpq0zpxZs/edit), and using it allows us to support xattrs in mac.
* Reuse as much code as possible between linux and macOS.

While this improves our support for macOS, in particular by supporting xattrs and avoiding macOS littering PFS with dotfiles, it does not fix the major standing issues documented in #4938 and #4982, which have proven to be hairier than I'd hoped. Investigation into that is still ongoing, but I figured this is worth merging in as-is since it'll likely be a while before I can circle back to those issues.

### CODE REVIEW

File: src/server/pfs/fuse/files_linux.go [hunk 1]
```diff
@@ -31,3 +32,13 @@ func (f *loopbackFile) utimens(a *time.Time, m *time.Time) syscall.Errno {
 	err := futimens(int(f.fd), &ts)
 	return fs.ToErrno(err)
 }
+
+// futimens - futimens(3) calls utimensat(2) with "pathname" set to null and
+// "flags" set to zero
+func futimens(fd int, times *[2]syscall.Timespec) (err error) {
```
Comment (jdoliner) [inline src/server/pfs/fuse/files_linux.go:38]: Named return params like this have bitten me a few times with shadowing so we generally only do it when you need to access the parameter in a deferred function. Could you switch this to just returning the error?

Comment (jdoliner) [reply src/server/pfs/fuse/files_linux.go:38]: Just realized this is copied from elsewhere so actually I was the one who checked this in originally >.< . (I was just copying it from the fuse library.) Given that I'm cool with leaving it as is.

**Comment by ysimonson** [comment]
Not sure if this is what you're referring to, but the `_darwin` and `_linux` filename suffixes act as compile guards

**Comment by jdoliner** [comment]
Ahh, that makes sense. I didn't realize go used filenames like that.

**Pull request merged**